### ISSUE TITLE
Allow for multiple common toolchain prefixes for a given target

### ISSUE THREAD

**Issue by xobs** [alexcrichton/cc-rs#519]
The `cc` crate assumes that there is one canonical name for each cross-compilation target. For example, `arm` seems to be usually called `arm-none-eabi` for embedded targets.

This doesn't hold up for all targets, where there are multiple compatible toolchain prefixes currently in use.

For example, the `riscv32i-unknown-none-elf` target can be compiled by three different common toolchains:

* `riscv32-unknown-elf`: The current default, and what you get if you compile a toolchain yourself from [the Risc-V project](https://github.com/riscv/riscv-tools) without multiarch support
* `riscv64-unknown-elf`: Probably the most popular prefix, which is available in Homebrew, [Arch AUR](https://aur.archlinux.org/packages/riscv64-unknown-elf-gcc/), and in binary form from [SiFive](https://www.sifive.com/boards/). This is what you get if you build `multiarch`. I'd wager most users have this.
* `riscv-non-embed`: From [xpack](https://xpack.github.io/riscv-none-embed-gcc/), which I'm only peripherally aware of and I gather is tuned for embedded usage

The `cc` project appears to only support a single default prefix per target:  https://github.com/alexcrichton/cc-rs/blob/77c94412c68bddd29d3ea86aa00c0aaecaaa4904/src/lib.rs#L2294-L2299

To use a toolchain that is not currently supported by `cc`, it is possible to set the `CC` environment variable to get `cc` to pick the correct toolchain. This is a fine workaround. Another option is to check in `.cargo/config` to set `CC`.

However, it would be nice if `cc` could have a list of common prefixes it could try as part of the build process.

**Comment by alexcrichton** [comment]
FWIW it's not really that this crate assumes one canonical name, it's typically just that a the names in Debian/Ubuntu are pretty common and no one's gotten around to implementing fancier logic. The purpose of the compiler detection is to find the most appropriate toolchain, but it's wrong lots of the time since it guesses executables that don't exist. It's always possible to enhance the detection here and implement something fancier for particular targets (e.g. actually probe the system and see what we find, or do what this issue says and search for a whitelist of prefixes)

Basically all that's to say I'd be happy to review a PR for this!

**Comment by Skallwar** [comment]
For clang at least you can do `riscv(32|64)XX-unknown-none-elf` => `riscv(32|64)-unknown-none-elf-riscv(32|64)XX`
